pkg/executor/default.go

```diff
@@ -102,14 +102,30 @@ func (e *DefaultExecutor) applyStage(stage schema.Stage, fs vfs.FS, console plug
 	return errs
 }
 
+func checkDuplicates(stages []schema.Stage) bool {
+	stageNames := map[string]bool{}
+	for _, st := range stages {
+		if _, ok := stageNames[st.Name]; ok {
+			return true
+		}
+		stageNames[st.Name] = true
+	}
+	return false
+}
+
 func (e *DefaultExecutor) genOpFromSchema(file, stage string, config schema.YipConfig, fs vfs.FS, console plugins.Console) []*op {
 	results := []*op{}
-
 	currentStages := config.Stages[stage]
 
+	duplicatedNames := checkDuplicates(currentStages)
+
 	prev := ""
 	for i, st := range currentStages {
 		name := st.Name
+		if duplicatedNames {
```

Comment:
Do we actually need to check or can we skip that and just add that index always and call it a day?

Reply:
@Itxaka yeah that was my first approach. I went with this because otherwise if you want to reference a step by name, you have to know in which position it will end. Thankfully there was a test so I saw this problem:

https://github.com/mudler/yip/blob/master/pkg/executor/default_test.go#L226

would need to be referenced as `test.0` which would be very confusing, specially if some step gets introduced before the wanted one